### DISCUSSION TITLE
[Issue] Coupon discount is applicable to all cart and not to its attached product(s) only

### DISCUSSION THREAD

**Discussion by josephelkhoury** [Blair2004/NexoPOS#1449] [Ideas]
**Describe the bug**
If you create a coupon with a discount code of 50% and set it as applicable on "Product A" only.

**To Reproduce**
Let's say you add "Product A" (10 USD) and "Product B" (5 USD) to the cart, the total is (without the coupon) 15 USD.
If you apply the coupon to the order, the total calculated is 7.5 USD (15 * 50%).

**Expected behavior**
The total should be 10 USD (10 * 50% + 5).

**Environment:**
 - NexoPOS 4.x version: 4.8.16

**Comment by Blair2004** [comment]
Hey,
Thank you for bringing this to our attention. In fact, when the coupon applies, it does to the cart. We haven't allowed coupons to apply to each product singularly, but obviously that's something we should consider.
